### DISCUSSION TITLE
How to mock featureCall with $cstNode?

### DISCUSSION THREAD

**Discussion by durianwaffle** [eclipse-langium/langium#1745] [Q&A]
I am currently using the code below to test a function `isFeatureCallArgNumberCorrect`. This function relays on `featureCall.$cstNode?.text` and also `featureCall.element.ref.type.reference?.ref.name`. 

The logic is to get the featurecall from parsed document. But I found that there is no $cstNode in the featurecall from parsed document. Why is it? Any suggestions for me to mock a featureCall that has those elements for my unit tests?
```TypeScript
let services = createStructuredTextServices(EmptyFileSystem);
let document: LangiumDocument<Model> | undefined;
let parse: ReturnType<typeof parseHelper<Model>>;
beforeAll(async () => {
    services = createXXServices(EmptyFileSystem);
    parse = parseHelper<Model>(services.StructuredText);
});

describe ((''), async() => {
document = await parse(`function_block TON()
        \tET: TIME;
        \tQ: BOOL;
        ;
        TON_1:TON;
        var_bool: bool;
        TON_1(var_bool);
        `);
        const items = collectTestItemsFromDocumentElements<FeatureCall>(document, FeatureCall);
        const res = isFeatureCallArgNumberCorrect(items[1]);}
});
```

**Comment by msujew** [comment]
Hey @durianwaffle,

All `AstNode` items should have a `$cstNode` assigned to them. Can you share a reproducible example so I can take a closer look at the issue?

**Reply by durianwaffle** [reply]
Hi @msujew thanks! Every item has $cstNode that's good.
I found the issue is because I used the 2nd time, I should pass in items[0]
